src/gui/application.cpp

```diff
@@ -674,4 +675,20 @@ void Application::slotGuiIsShowingSettings()
     emit isShowingSettingsDialog();
 }
 
+bool Application::eventFilter(QObject *watched, QEvent *event)
+{
+    if (watched && watched->isWidgetType()
+        && (event->type() == QEvent::Show || event->type() == QEvent::Hide)) {
+        auto dialog = qobject_cast<QDialog*>(watched);
+        const auto visible = (event->type() == QEvent::Show);
+
+        // Let ownCloudGui manage visible dialogs
+        if (_gui) {
```

Comment:
Wondering if there are chance of this ever be false, we always create windows after that _gui object anyway? If I'm right I'd turn that assumption into a Q_ASSERT instead.

Reply:
Should be that way, I've now also moved the `installEventFilter` below the `_gui` initialization.